### DISCUSSION TITLE
DSC Unlocker 1.1 Nodemcu

### DISCUSSION THREAD

**Discussion by BigHomie90** [taligentx/dscKeybusInterface#167] [dscKeybusInterface Discussion]
Hello! First of all I want to thank you for your code!
I am trying to find my installer code on a DSC pc1616.
Keypad lockout is enabled after 12 attempts and it unlocks again in approximately 130 sec. When I power on the unit the phone relay is clicking 8-10 times so I cannot by pass the keypad lockout by shorting pgm1 and zone1. So far I made the wiring according to the DSC Unlocked sketch and I have opened the terminal to see the logs. If I read correctly the process stops automatically when the code is correct and the nodemcu led blinks according to code. Is there anyone who can post a picture/screenshot on how the serial monitor displays the password when found? Till now it tested 1500+ codes as you can see on my picture. I just want to be sure that if it finds the code how it will show up in the serial window. Thank you!
![IMG_20201106_011013](https://user-images.githubusercontent.com/59261389/98306615-da3aad00-1fcc-11eb-8fdb-2ce397388328.jpg)

**Comment by taligentx** [comment]
Hi @BigHomie90 - great to see the Unlocker sketch being put to use!

>  If I read correctly the process stops automatically when the code is correct and the nodemcu led blinks according to code. Is there anyone who can post a picture/screenshot on how the serial monitor displays the password when found? 

I don't have a screenshot, but for example if the installer code is 0878, the output should look like:
```
12921s | Test position: 1645 | Test code: 0878
12922s | Installer code: 0878
```
Worst case, the sketch should run through all codes on your panel in ~22 hours - you should also see the keypads return to normal once the code search wraps up.  If you can grab a screenshot once the code search completes, I can add it as a text example to the sketch - thanks!

-Nikhil

**Reply by Weezerxiv14** [reply]
Hi @BigHomie90 . Glad to see the unlocker did the work for your dsc pc1616. For now I'm unable to get it work on pc1616 with a arduino uno R3, even with a mega2560. All my wirings are ok. Did you changed something in the sketche to get it works?  It is supposed to run automatically after upload to the board. Wich board did you use? Any help would be very appreciated.

**Reply by BigHomie90** [reply]
Hey @Weezerxiv14 . It's quite a while i did this but i am sure that i didn't change anything in arduino sketch. I used an esp8266 (cp2102).
1.Double check your wiring (resistors etc.)
2.Upload the sketch to your arduino board
3. while connected to the alarm system just reboot the arduino and open serial output to see what is it doing.
In my case, it auto started. Also be sure that alarm system is in disarmed state.

*I was powering my esp8266 from a different power supply (5v) but ground was common between esp8266 and alarm panel.
**Also check the baud rate according to instructions. I don't remember the exact value. 9600 or 11520 i suppose it was.

**Comment by BigHomie90** [comment]
If the script finds the code of course I will upload the screenshot! Also if I don't get the code for whatever reason I will post exactly what I have tried so far in case it could be useful to someone. I ll post again when the process finishes! (So far 3070 codes tested and still running)

**Reply by taligentx** [reply]
> Also if I don't get the code for whatever reason I will post exactly what I have tried so far in case it could be useful to someone. I ll post again when the process finishes! 

Sounds good, and good luck!  It'll be especially interesting if the code isn't found, it'd be handy if you can keep a copy of the entire serial log to help troubleshoot if this happens.

**Reply by BigHomie90** [reply]
I want to thank you so much!!!!
The code found! 0154
I just want to put a note that serial monitor didn't show ap anything when it found the code. Every 12 code attempts I got used to see a keypad locked. Suddenly I saw 6 codes and nothing after that. So I thought at first that the node mcu hanged and restarted it to check the same codes again.It was stuck again. I tried to start the sketch after the last code was shown and it started normally so I entered the last code (0154) shown on the serial and it was correct!

**Comment by BigHomie90** [comment]
![IMG_20201106_193928](https://user-images.githubusercontent.com/59261389/98397402-f93a4d00-2067-11eb-94aa-791ceefa3921.jpg)

**Comment by taligentx** [comment]
> I just want to put a note that serial monitor didn't show ap anything when it found the code. Every 12 code attempts I got used to see a keypad locked. Suddenly I saw 6 codes and nothing after that. So I thought at first that the node mcu hanged and restarted it to check the same codes again.It was stuck again. I tried to start the sketch after the last code was shown and it started normally so I entered the last code (0154) shown on the serial and it was correct!

Great to hear you found the code!  But sounds like you've run into a bug in the Unlocker sketch, I'd like to see why it stopped after finding the code and before outputting the code by serial and blinking the LED - there may be some difference with the PC1616.  If you're able to grab some logs to fix this bug, I'd like to see the output of the `KeybusReader` sketch:

1. Start the `KeybusReader` sketch and start capturing the serial output while the panel is disarmed and not in programming mode - let it capture data in this mode for 5 minutes.
2. Enter *8 programming with a wrong installer code either on a physical keypad or through the serial terminal input, then press # to get back to normal mode.
3.  Wait 30 seconds or so, then enter 8* programming with the correct installer code.
4. Capture another 30 seconds or so of logs while in installer mode, and then press # to exit programming.
5. Capture a final 30 seconds or logs while back in normal mode.

This should help narrow down exactly why the sketch failed to output the code - if you're up for capturing this log, please open a new Issue to help track the bug.  Thanks for the feedback and screenshot!

**Reply by BigHomie90** [reply]
I opened an issue about this but because i dont know how to extract logs from serial monitor to txt or something like that, i uploaded 3 videos on youtube and posted the links. Tomorrow i will try to make the post "cleaner" and post again.

**Comment by rionel2** [comment]
Hola buenas noches queria consultar si el bloqueo del instalador esta habilitado ya sea x dls5 o teclado con este circuito se logra descifrar igual la clave ? xq no quisiera comprar el esp8266 y que no me sirva al menos para este caso de poder sacar el codigo de instalador....gracias
